src/main/java/com/stud/java/entityGenerators/BaseEntityGenerator.java

```diff
@@ -0,0 +1,37 @@
+package com.stud.java.entityGenerators;
+
+import com.stud.java.repository.entity.BaseEntity;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+
+public abstract class BaseEntityGenerator {
```

Comment:
Интересное решение)

Раз уж взялся за наследование, то давай сделаем прям круто

Смени название на EntityGenerator и используй дженерик <T extends BaseEntity>, чтобы получилось
```suggestion
public abstract class EntityGenerator<T extends BaseEntity> {
```

Reply:
С тем решением, что есть сейчас у нас есть ограничение по возвращаемому списку 
Он всегда будет в себе держать BaseEntity, что некорректно 
List<BaseEntity> != List<CatEntity>

Чтобы получилось и наследование и корректный результат, нужно делать через дженерики